docs/designs/conceptual-versioning.md

```diff
@@ -0,0 +1,395 @@
+
+we can get an ordered moniker list by the API set in config as `monikerDefinition`, the response should be:
+
+```json
+{
+    "monikers":[
+        {
+            "monikerName": "{monikerName}",
+            "productName": "{productName}",
+            "order" : {order}
+        },
+        ...
+    ]
+}
+```
+
+1. `monikerName` is an unique identifier of this moniker. If two moniker define the same `monikerName`, an error throws.
+
+2. `productName` defines the product this moniker belongs to, and the operators in monikerRange will be interpreted inside the product it belongs to.
+
+3. `order` defines the order of this moniker in its product. If two moniker with same `productName` define the same `order`, an error throws.
+
+### 4.2 Empty page
+
+When there is a file contains a moniker, but no content is included in this moniker, there will be a empty page.
+
+Sample:
+
+```markdown
+---
+monikerRange: netcore-1.0 || netcore-2.0
+---
+
+# Title
+
+::: moniker range="netcore-1.0"
+content in netcore-1.0
+::: moniker-end
+```
+
+In this case, there is no content with moniker `netcore-2.0`, when user view this page with `?view=netcore-2.0`, they should be fallback to `netcore-1.0` but not getting a empty page.
+
+To handle this case, we have to:
+
+1. Remove this moniker from the moniker range from this file's build result.
+1. Remove this moniker from the moniker range of this file in the Toc.
+1. Remove this moniker from the moniker range of this file in the Xref map.
+
+After doing doing this,
+
+1. User can not find this file in the Toc when they select the moniker of Toc as `netcore-2.0`.
+1. If user access url `{page_url}?view=netcore-2.0`, the DHS will fallback to `{page_url}?netcore-1.0`
+
+### 4.3 Reference resolve
+
+#### 4.3.1 Link resolve
+
+In phase 1, link will be resolved to relative sitePath without version info, DHS will handle the fallback behavior if the referenced page doesn't the version of current page.
+
+#### 4.3.2 Xref resolve
+
+In v2, each group will build one round, and in each group, there should be not be two files with the same uid, so the xref can be resolved correctly.
+But in v3, there is no group, all files will be built in one round, so there will be two file with same uid but different version, which is not acceptable, and we have to handle this case.
+To handle this, when we are generating the xref map, we have to contains the monikers of each uid.
+
+If in one round of build, different files with the same **Uid** are included:
+
+1. When the files do not have `monikerRange` option set, an error throws.
+2. When the files have `monikerRange` option set, but have different **SitePath**, an error throws.
+3. When the files have `monikerRange` option set, and have the same **SitePath**, these file are considered as different version of the same **Uid**, this is allowed.
+
+New xrefMap model should be
```

Comment:
Basically for internal xref resolving, moniker range will be involved, but not external xref, so xref map output for external usage would not contain moniker range info.
Think we can define the exact contract while we start the implementation since it might be changed.